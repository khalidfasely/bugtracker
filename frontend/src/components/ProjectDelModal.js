import Modal from "react-modal";

const ProjectDelModal = ({ delModalOpen, setDelModalOpen, deleteProject }) => (
    <Modal
        className='delete-modal'
        isOpen={delModalOpen}
        onRequestClose={() => setDelModalOpen(false)}
    >
        <p className="modal__paragraph">If you delete this project there is no way to return it!</p>
        <button className="modal__del-button" onClick={deleteProject} data-testid='delete_button'>Delete</button>
    </Modal>
);

export default ProjectDelModal;