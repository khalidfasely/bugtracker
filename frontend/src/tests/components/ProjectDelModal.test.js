import React from 'react';
import { fireEvent, render } from "@testing-library/react";

import ProjectDelModal from '../../components/ProjectDelModal';

let props;

beforeEach(() => {
    props = {
        delModalOpen: true,
        setDelModalOpen: jest.fn(),
        deleteProject: jest.fn()
    };
});

test('Should render ProjectDelModal component correctly', () => {
    const { asFragment } = render(<ProjectDelModal {...props} />);
    expect(asFragment()).toMatchSnapshot();
});

test('Should render/fire delete button correctly', () => {
    const { getByTestId } = render(<ProjectDelModal {...props} />);
    const deleteButtonEl = getByTestId('delete_button');

    expect(deleteButtonEl.textContent).toBe('Delete');

    expect(props.deleteProject).toHaveBeenCalledTimes(0);

    fireEvent.click(deleteButtonEl);

    expect(props.deleteProject).toHaveBeenCalledTimes(1);
});