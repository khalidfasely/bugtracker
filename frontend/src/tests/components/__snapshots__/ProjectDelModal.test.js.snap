// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ProjectDelModal component correctly 1`] = `
<DocumentFragment>
  <p
    class="modal__paragraph"
  >
    If you delete this project there is no way to return it!
  </p>
  <button
    class="modal__del-button"
    data-testid="delete_button"
  >
    Delete
  </button>
</DocumentFragment>
`;
